core/src/main/java/tc/oc/pgm/stats/PlayerStats.java

```diff
@@ -0,0 +1,188 @@
+import net.kyori.text.Component;
+import net.kyori.text.TextComponent;
+import net.kyori.text.TranslatableComponent;
+import net.kyori.text.format.TextColor;
+
+/** A wrapper for stat info belonging to a {@link tc.oc.pgm.api.player.MatchPlayer} */
+class PlayerStats {
+  // K/D
+  private int kills;
+  private int deaths;
+  private int killstreak; // Current killstreak
+  private int killstreakMax; // The highest killstreak reached this match
+
+  // Bow
+  private int longestBowKill;
+  private double bowDamage;
+  private int shotsTaken;
+  private int shotsHit;
+
+  // Damage
+  private double damageDone;
+  private double damageTaken;
+
+  // Objectives
+  private int destroyablePiecesBroken;
+  private int flagsCaptured;
+
+  private Duration longestFlagHold = Duration.ZERO;
+  private Instant longestFlagHoldCache;
+
+  // The task responsible for displaying the stats over the hotbar
+  // See StatsMatchModule#sendLongHotbarMessage
+  private Future<?> hotbarTaskCache;
+
+  // Methods to update the stats, should only be accessed by StatsMatchModule
+
+  protected void onMurder() {
+    kills++;
+    killstreak++;
+    if (killstreak > killstreakMax) killstreakMax = killstreak;
+  }
+
+  protected void onDeath() {
+    deaths++;
+    killstreak = 0;
+  }
+
+  protected void onDamage(double damage, boolean bow) {
+    damageDone += damage;
+    if (bow) {
+      bowDamage += damage;
+      shotsHit++;
+    }
+  }
+
+  protected void onDamaged(double damage) {
+    damageTaken += damage;
+  }
+
+  protected void onDestroyablePieceBroken(int change) {
+    destroyablePiecesBroken += change;
+  }
+
+  protected void onFlagCapture() {
+    flagsCaptured++;
+    onFlagDrop();
+  }
+
+  protected void onFlagPickup() {
+    longestFlagHoldCache = Instant.now();
+  }
+
+  protected void onFlagDrop() {
+    setLongestFlagHold(
+        Duration.ofMillis(Instant.now().toEpochMilli() - longestFlagHoldCache.toEpochMilli()));
+  }
+
+  protected void setLongestFlagHold(Duration time) {
+    if (longestFlagHold == null || (time.toNanos() - longestFlagHold.toNanos()) > 0)
+      longestFlagHold = time;
+  }
+
+  protected void setLongestBowKill(double distance) {
+    if (distance > longestBowKill) {
+      longestBowKill = (int) Math.ceil(distance);
+    }
+  }
+
+  protected void onBowShoot() {
+    shotsTaken++;
+  }
+
+  // Makes a simple stat message for this player that fits in one line
+
+  public Component getBasicStatsMessage() {
+    return TranslatableComponent.of(
+        "match.stats",
+        TextColor.GRAY,
+        TextComponent.of(kills, TextColor.GREEN),
+        TextComponent.of(killstreak, TextColor.GREEN),
+        TextComponent.of(deaths, TextColor.RED),
+        TextComponent.of(getKD(), TextColor.GREEN));
+  }
+
+  // Getters, both raw stats and some handy calculations
+
+  public double getKD() {
+    double kd;
+    if (deaths == 0) {
+      kd = kills;
+    } else {
+      kd = kills / (double) deaths;
+    }
+    return kd;
```

Comment:
```suggestion
return kills / Math.max(1d, (double) deaths);
```
Simplify this, you may be able to get rid of the double cast too if the first parameter is explicitly double

Reply:
It will(is what my IDE says)! Since `Math.max` returns a double in that case